.github/workflows/pr_metadata.yml

```diff
@@ -0,0 +1,27 @@
+# Auto-update metadata on pull requests
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+name: "Pull Request Metadata"
+on:
+- pull_request_target
```

Comment:
This will re-add the labels and re-assign the author each time a commit is pushed, but I'm not sure if this is intentional. For example maybe the PR author doesn't think a tag is appropriate and removes it, but it's automatically added on the next commit or rebase. Or if the author doesn't want to be assigned to a PR.

Maybe restrict it to:

```yaml
on:
  pull_request_target:
    types: [opened]
```

Reply:
Done in 6364837

Reply:
This means if you push more commits to the PR later that change files in a new component, the new component tag will not be automatically added.